src/NHibernate/Util/ADOExceptionReporter.cs

```diff
@@ -19,20 +19,25 @@ public static void LogExceptions(Exception ex)
 
 		public static void LogExceptions(Exception ex, string message)
 		{
-			if (log.IsErrorEnabled)
+			if (log.IsErrorEnabled())
 			{
-				if (log.IsDebugEnabled)
+				if (log.IsDebugEnabled())
 				{
 					message = StringHelper.IsNotEmpty(message) ? message : DefaultExceptionMsg;
-					log.Debug(message, ex);
+					log.Debug(ex, message);
 				}
+
+				// Pass full exception on highest call
+				if (log.IsWarnEnabled()) log.Warn(ex, ex.ToString());
+				log.Error(ex, ex.Message);
+				ex = ex.InnerException;
```

Comment:
Why those changes and the next?
That changes what is was actually doing. Previously if warn was enabled it would have fully logged inner exceptions. Now it does no more.